File: ios/ios/Models/JenkinsView.swift
//
//  JenkinsView.swift
//  ios
//
//  Jenkins view model (not to be confused with SwiftUI View)
//

import Foundation

struct JenkinsView: Codable, Identifiable, Equatable {
    let name: String
    let url: String
    let jobs: [Job]?
    
    var id: String { name }
    
    var displayName: String {
        if name == "all" || name == "All" {
            return "所有"
        }
        return name
    }
}

// MARK: - API Response Wrappers

struct JenkinsRootResponse: Codable {
    let views: [JenkinsView]
    let primaryView: JenkinsView?
    let nodeDescription: String?
    let nodeName: String?
    let mode: String?
}

struct ViewDetailResponse: Codable {
    let name: String
    let url: String
    let jobs: [Job]
}

File: android/app/src/main/java/com/by/android/data/model/JenkinsView.kt
package com.by.android.data.model

data class JenkinsView(
    val name: String,
    val url: String,
    val jobs: List<Job>? = null
) {
    val displayName: String
        get() = if (name.equals("all", ignoreCase = true)) "所有" else name
}

// API Response wrappers
data class JenkinsRootResponse(
    val views: List<JenkinsView>,
    val primaryView: JenkinsView? = null,
    val nodeDescription: String? = null,
    val nodeName: String? = null,
    val mode: String? = null
)

data class ViewDetailResponse(
    val name: String,
    val url: String,
    val jobs: List<Job>
)

